### PR TITLE
OCPBUGSM-18042 Agent unable to start container during discovery

### PR DESCRIPTION
**Pull request by ori-amizur**
- Add timeout of 300 seconds by default to wait for image pull
/cc @ronniel1

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -85,6 +85,7 @@ type Config struct {
 	SkipCertVerification bool              `envconfig:"SKIP_CERT_VERIFICATION" default:"false"`
 	InstallRHCa          bool              `envconfig:"INSTALL_RH_CA" default:"false"`
 	RhQaRegCred          string            `envconfig:"REGISTRY_CREDS" default:""`
+	AgentTimeoutStartSec int64             `envconfig:"AGENT_TIMEOUT_START_SEC" default:"300"`
```
Comment (ronniel1) [inline internal/bminventory/inventory.go:88]: This should be time.Duration with a default of "3m"
See ImageExpirationTime variable above

**Review by ronniel1 — COMMENTED** [review]
Need to verify that adding the `TimeoutStartSec` helps in the case that `ExecStartPre` is slow

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (ori-amizur) [reply internal/bminventory/inventory.go:88]: OK

Comment (ronniel1) [reply internal/bminventory/inventory.go:88]: This is the way that we specify timeouts to the asssited-service
See how `AgentTimeoutStartSec` is used

**Review by gamli75 — APPROVED** [review]

**Pull request merged**